lib/ansible/modules/cloud/spotinst/spotinst_aws_elastigroup.py

```diff
@@ -0,0 +1,1509 @@
+        health_check_unhealthy_duration_before_replacement=dict(type='int'),
+        iam_role_arn=dict(type='str'),
+        iam_role_name=dict(type='str'),
+        image_id=dict(type='str', required=True),
+        key_pair=dict(type='str'),
+        kubernetes=dict(type='dict'),
+        lifetime_period=dict(type='int'),
+        load_balancers=dict(type='list'),
+        max_size=dict(type='int', required=True),
+        mesosphere=dict(type='dict'),
+        min_size=dict(type='int', required=True),
+        monitoring=dict(type='str'),
+        multai_load_balancers=dict(type='list'),
+        multai_token=dict(type='str'),
+        name=dict(type='str', required=True),
+        network_interfaces=dict(type='list'),
+        on_demand_count=dict(type='int'),
+        on_demand_instance_type=dict(type='str'),
+        opsworks=dict(type='dict'),
+        persistence=dict(type='dict'),
+        product=dict(type='str', required=True),
+        rancher=dict(type='dict'),
+        right_scale=dict(type='dict'),
+        risk=dict(type='int'),
+        roll_config=dict(type='dict'),
+        scheduled_tasks=dict(type='list'),
+        security_group_ids=dict(type='list', required=True),
+        shutdown_script=dict(type='str'),
+        signals=dict(type='list'),
+        spin_up_time=dict(type='int'),
+        spot_instance_types=dict(type='list', required=True),
+        state=dict(default='present', choices=['present', 'absent']),
+        tags=dict(type='list'),
+        target=dict(type='int', required=True),
+        target_group_arns=dict(type='list'),
+        tenancy=dict(type='str'),
+        terminate_at_end_of_billing_hour=dict(type='bool'),
+        unit=dict(type='str'),
+        user_data=dict(type='str'),
+        utilize_reserved_instances=dict(type='bool'),
+        uniqueness_by=dict(default='name', choices=['name', 'id']),
+        up_scaling_policies=dict(type='list'),
+        target_tracking_policies=dict(type='list'),
+        wait_for_instances=dict(type='bool', default=False),
+        wait_timeout=dict(type='int')
+    )
+
+    module = AnsibleModule(argument_spec=fields)
+
+    if not HAS_SPOTINST_SDK:
+        module.fail_json(msg="the Spotinst SDK library is required. (pip install spotinst)")
+
+    # Retrieve creds file variables
+    creds_file_loaded_vars = dict()
+
+    credentials_path = module.params.get('credentials_path')
+
+    with open(credentials_path, "r") as creds:
+        for line in creds:
+            eq_index = line.find('=')
+            var_name = line[:eq_index].strip()
+            string_value = line[eq_index + 1:].strip()
+            creds_file_loaded_vars[var_name] = string_value
+    # End of creds file retrieval
+
+    token = os.environ.get('SPOTINST_TOKEN')
+    if not token:
+        token = creds_file_loaded_vars.get("token")
+
+    account = os.environ.get('ACCOUNT')
+    if not account:
+        account = creds_file_loaded_vars.get("account")
```

Comment:
This would put the account in the creds file *above* the params directly passed to the module. This should probably be reversed, as we try to follow the order of specificity (preferring most specific) as: /etc file, $HOME file, environment variable, direct specification on task parameter.